notifier/processor_create_test.go

```diff
@@ -2,260 +2,331 @@ package notifier
 
 import (
 	"context"
-	"fmt"
+	"errors"
+	"strconv"
+	"sync"
 	"testing"
+	"time"
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/google/uuid"
-	"github.com/quay/clair/v4/indexer"
-	"github.com/quay/clair/v4/matcher"
 	"github.com/quay/claircore"
 	"github.com/quay/claircore/libvuln/driver"
+
+	"github.com/quay/clair/v4/indexer"
+	"github.com/quay/clair/v4/matcher"
 )
 
-var (
-	vulnAdd = &claircore.Vulnerability{
-		ID:          "0",
-		Name:        "added vulnerability",
-		Description: "a vulnerability added",
-	}
-	vulnRemoved = &claircore.Vulnerability{
-		ID:          "1",
-		Name:        "removed vulnerability",
-		Description: "a vulnerability removed",
-	}
-	manifestAdd          = `sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef`
-	manifestRemoved      = `sha256:fc92eec5cac70b0c324cec2933cd7db1c0eae7c9e2649e42d02e77eb6da0d15f`
-	affectedManifestsAdd = &claircore.AffectedManifests{
-		Vulnerabilities: map[string]*claircore.Vulnerability{
-			vulnAdd.ID: vulnAdd,
-		},
-		VulnerableManifests: map[string][]string{
-			manifestAdd: []string{vulnAdd.ID},
-		},
-	}
-	affectedManifestsRemoved = &claircore.AffectedManifests{
-		Vulnerabilities: map[string]*claircore.Vulnerability{
-			vulnRemoved.ID: vulnRemoved,
-		},
-		VulnerableManifests: map[string][]string{
-			manifestRemoved: []string{vulnRemoved.ID},
-		},
-	}
-	notifications = []Notification{
-		{
-			Manifest: claircore.MustParseDigest(manifestAdd),
+// ProcessorTestCase handles all the stuff for a Processor test case.
+//
+// Any exported member will have defaults generated except for Name.
+type processorTestCase struct {
+	// Needed arguments.
+	Name string
+	// Optional arguments.
+	NoSummary   bool
+	Err         bool
+	PrevUUID    uuid.UUID
+	UpdaterName string
+	OpRef       uuid.UUID
+	EventDate   time.Time
+	// These functions are like the Named methods on various interfaces, with
+	// the addition of a testing.T object for convenience.
+	UpdateDiff        func(*testing.T, context.Context, uuid.UUID, uuid.UUID) (*driver.UpdateDiff, error)
```

Comment:
this is conflating the use of a mock, and a test case. I think its confusing to have them. this seems like a use of "clever" over "obvious"